test/v1alpha1/route.go

```diff
@@ -102,13 +102,15 @@ func CheckRouteState(client *test.ServingAlphaClients, name string, inState func
 // IsRouteReady will check the status conditions of the route and return true if the route is
 // ready.
 func IsRouteReady(r *v1alpha1.Route) (bool, error) {
-	return r.Generation == r.Status.ObservedGeneration && r.Status.IsReady(), nil
+	return r.IsReady(), nil
 }
 
-// IsRouteNotReady will check the status conditions of the route and return true if the route is
+// IsRouteFailed will check the status conditions of the route and return true if the route is
 // not ready.
-func IsRouteNotReady(r *v1alpha1.Route) (bool, error) {
-	return !r.Status.IsReady(), nil
+func IsRouteFailed(r *v1alpha1.Route) (bool, error) {
+	rs := r.Status
+	return rs.ObservedGeneration == r.Generation &&
+		r.GetConditionSet().Manage(&rs).GetTopLevelCondition().IsFalse(), nil
```

Comment:
Let's not expose the condition set and just add a helper to the resource ie. `IsFailed()`